scripts/test_reports/update_testsuite.py

```diff
@@ -0,0 +1,272 @@
+            subsection = row[1]
+            print("Subsection: ", subsection)
+            if sec_id:
+                subsec_id = run.find_section(sections, subsection, sec_id)
+                print("subsec_id: ", subsec_id)
+
+                if not subsec_id:
+                    print(" ==> No subsection found. Creating {}/{}".format(section, subsection))
+                    run.add_subsection(sections, section, sec_id, subsection)
+                    sections = run.client.section.for_suite(run.project_id,
+                                                            suite_id=run.suite)
+                    subsec_id = run.find_section(sections, subsection, sec_id)
+                    print("=> Adding testcase: ", testcase_name)
+                    run.add_testcase(subsec_id, testcase_name)
+                    testcases_counter += 1
+
+                print("=> Adding testcase: ", testcase_name)
+                run.add_testcase(subsec_id, testcase_name)
+                testcases_counter += 1
+            else:
+                print(" ==> No section found. Creating {}/{}".format(section, subsection))
+                run.add_section(section)
+                sections = run.client.section.for_suite(run.project_id,
+                                                        suite_id=run.suite)
+                sec_id = run.find_section(sections, section)
+                run.add_subsection(sections, section, sec_id, subsection)
+                sections = run.client.section.for_suite(run.project_id,
+                                                        suite_id=run.suite)
+                subsec_id = run.find_section(sections, subsection, sec_id)
+                print("=> Adding testcase: ", testcase_name)
+                run.add_testcase(subsec_id, testcase_name)
+                testcases_counter += 1
+        else:
+            #That branch will run only if testcase
+            #has only a section name without testcase name at all
+            #It means that testcase consists only from one word
+            #For that kind of test case is unable to detect subsection,
+            #only section can be detected
+            if not sec_id:
+                print(" ==> No section found. Creating {}".format(section))
+                run.add_section(section)
+                sections = run.client.section.for_suite(run.project_id,
+                                                        suite_id=run.suite)
+                sec_id = run.find_section(sections, section)
+                print("=> Adding testcase: ", testcase_name)
+                run.add_testcase(sec_id, testcase_name)
+                testcases_counter += 1
+
+            print("=> Adding testcase: ", testcase_name)
+            run.add_testcase(sec_id, testcase_name)
+            testcases_counter += 1
+
+def main():
+    args = parse_args()
+    run = TestSuiteUpdate(results_file=args.results_file, project_id=args.project, suite=args.suite)
```

Comment:
You need to check for the arguments before you use them, right now if I run the script without any arguments I get:

```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): zephyrproject.testrail.io:443
DEBUG:urllib3.connectionpool:https://zephyrproject.testrail.io:443 "GET /index.php?/api/v2/get_cases/None&suite_id=None&sectio8
Traceback (most recent call last):
  File "scripts/test_reports/update_testsuite.py", line 271, in <module>
    main()
  File "scripts/test_reports/update_testsuite.py", line 242, in main
    cases_suite = run.receive_cases_suite()
  File "scripts/test_reports/update_testsuite.py", line 46, in receive_cases_suite
    cases_suite = self.client.case.for_project(project_id=self.project_id, suite_id=self.suite)
  File "/Users/anashif/zephyr/zephyrproject/qm/scripts/test_reports/testrail_client/api/case.py", line 37, in for_project
    params=filters)
  File "/Users/anashif/zephyr/zephyrproject/qm/scripts/test_reports/testrail_client/api/base.py", line 49, in _format
    return func(*args, **kwargs)
  File "/Users/anashif/zephyr/zephyrproject/qm/scripts/test_reports/testrail_client/api/base.py", line 35, in _check
    raise TesRailAPIError(resp)
testrail_client.api.error.TesRailAPIError: [TestRailAPI] 400 || (Bad Request) || b'{"error":"Field :project_id is not a valid '
```